CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8.c

```diff
@@ -40,46 +40,320 @@
  * @{
  */
 
-  /*
-   *  Basic s8 depthwise convolution function.
-   *
-   *  Refer header file for details.
-   *  Optimization using DSP extension is not available for the generic case where channel multiplier is > 1.
-   *
-   */
-arm_status arm_depthwise_conv_s8(const q7_t *input,
-                                 const uint16_t input_x,
-                                 const uint16_t input_y,
-                                 const uint16_t input_ch,
-                                 const q7_t *kernel,
-                                 const uint16_t output_ch,
-                                 const uint16_t ch_mult,
-                                 const uint16_t kernel_x,
-                                 const uint16_t kernel_y,
-                                 const uint16_t pad_x,
-                                 const uint16_t pad_y,
-                                 const uint16_t stride_x,
-                                 const uint16_t stride_y,
-                                 const int32_t *bias,
-                                 q7_t *output,
-                                 const int32_t *output_shift,
-                                 const int32_t *output_mult,
-                                 const uint16_t output_x,
-                                 const uint16_t output_y,
-                                 const int32_t output_offset,
-                                 const int32_t input_offset,
-                                 const int32_t output_activation_min,
-                                 const int32_t output_activation_max,
-                                 const uint16_t dilation_x,
-                                 const uint16_t dilation_y,
-                                 q15_t *buffer_a)
+__STATIC_FORCEINLINE void depthwise_conv_s8_mult_1(const q7_t *input,
```

Comment:
In order to decrease the code size, you can remove the inline keyword. Since it is called only once, it should not be a big problem in terms of performance

Reply:
I would put "static" in order to avoid to expose it externally. in this manner you can change the interface without any problems (uint16_t -> int32_t)